event.go

```diff
@@ -34,6 +35,7 @@ type EventDispatcher interface {
 type DefaultEventHandler struct {
 	dispatcher             EventDispatcher
 	stash                  *stash
+	stashCache             *stash
```

Comment:
The `stash` has to support multiple blocks because we don't wait for events to be emitted before moving onto the next block, the cache on the other hand should only hold events for the current tx.

Reply:
ok fixed